eclipse-che/che-operator
# feat: Syncing of proxy settings and self-signed cert to the user's workspace namespace

### What does this PR do?
This adds the ability to sync settings to the workspace namespaces of the users.

The new controller watches namespaces (or projects on OpenShift). If a namespace has appropriate label (`org.eclipse.che/workspace-namespace-owner-uid`), it is considered a user workspace namespace and is synced to.

The value of the label is supposed to be the UID of the user. On OpenShift, this is the UID of the user object and on Kubernetes, the UID of the user as obtained from the OAuth token of the configured OIDC provider.

### What issues does this PR fix or reference?
eclipse/che#20168

### How to test this PR?
create a namespace, label it and check that the proxy settings are synced to it in a config map and self signed certificates in a secret.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
